src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java

```diff
@@ -339,6 +341,80 @@ public static String getVulnerabilityInfoString(Traces traces) {
         }
 
     }
+    /**
+     * Collection of all traces for an application
+     *
+     * @param sdk ContrastSDK instance
+     * @param organizationId Organization ID of the application
+     * @param applicationId Application ID
+     * @param filter TraceFormFilter to limit results
+     * @return Traces object
+     */
+    public static Traces getAllApplicationTraces(ContrastSDK sdk, String organizationId, String applicationId, TraceFilterForm filter) throws IOException, UnauthorizedException {
+        Traces traces = new Traces();
+
+        // Contrast API returns traces paginated at 20 per page by default. Increase page size for faster transfer.
+        int page = 0;
+        int pageSize = 50;
+        filter.setLimit(pageSize);
+
+        do {
+            filter.setOffset(page * pageSize);
+
+            try {
+                Traces intermediateTraces = sdk.getTraces(organizationId, applicationId, filter);
+
+                if (page == 0) {
+                    traces = intermediateTraces;
+                } else {
+                    traces.getTraces().addAll(intermediateTraces.getTraces());
+                }
+            } catch (IOException | UnauthorizedException e) {
+                throw e;
+            }
+
+            page++;
+        } while (traces.getTraces().size() < traces.getCount());
```

Comment:
So only 20 traces are returned from the API, but the count is the total number?

Reply:
Correct! If the filter doesn't specify a limit, TeamServer returns 20 traces. `traces.getCount()` gets the value as reported by TeamServer in the response object.